cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -176,36 +185,52 @@ static Bom createBom() {
         return bom;
     }
 
-    static void signSBOM(final String keystoreFile, final String keystorePassword, final String privateKey, final String publicKey) {
+    static void signBOM() {
         try {
-            // Load the keystore
-            JSONAsymKeySigner signer = new JSONAsymKeySigner(new FileInputStream(keystoreFile), keystorePassword.toCharArray(), privateKey, publicKey);
-            System.out.println("Keystore loaded successfully.");
-
-            File homeDir = new File(System.getProperty("user.home"));
-            File[] directoryListing = homeDir.listFiles();
-            if (directoryListing != null) {
-                for (File sbomFile : directoryListing) {
-                    if (sbomFile.getName().endsWith(".json")) {
-                        // read the sbom file as a json object writer
-                        JSONObjectWriter sbom = new JSONObjectWriter(new FileInputStream(sbomFile));
-                        // sign the sbom
-                        byte[] signedSbom = signer.signData(sbom.getEncoded());
-                        // Write the signed SBOM to a new file
-                        FileOutputStream signedSbomStream = new FileOutputStream("signed_" + sbomFile.getName());
-                        signedSbomStream.write(signedSbom);
-                        signedSbomStream.close();
-                        System.out.println("SBOM " + sbomFile.getName() + " signed and saved as signed_" + sbomFile.getName());
-                    }
-                }
-            } else {
-                System.out.println("Error: the home directory is empty or does not exist.");
-                System.exit(1);
-            }
-            // Print the certificate information
-            System.out.println("SBOMs signed with certificate: " + signer.getCertificate().toString());
-        } catch (Exception e) {
-            System.out.println("Error signing SBOM: " + e.getMessage());
+            // Read multiple JSON files to be signed
+            File[] jsonFiles = new File[n]; // where n is the number of files to be read
+            byte[][] jsonData = new byte[n][];
+            String[] sbomDataToSign = new String[n];
+            for (int i = 0; i < n; i++) {
+            FileInputStream jsonFileInputStream = new FileInputStream(jsonFiles[i]);
+            jsonData[i] = new byte[(int) jsonFiles[i].length()];
+            jsonFileInputStream.read(jsonData[i]);
+            jsonFileInputStream.close();
+            sbomDataToSign[i] = new String(jsonData[i], "UTF-8");
+        }
+
+            // Read the private key
+            File privateKeyFile = new File("RSAPrivate.pem");
```

Comment:
Needs to read the file passed in pemFile, which is the parameter to the --signSBOM operation.